Update consumer_ccsr.py

ecukuroglu opened the pull request:
The ordering of the variables is fixed to prevent deserialisation error.

### Description 

<!-- https://confluentinc.atlassian.net/browse/DEVX- -->

_What behavior does this PR change, and why?_


### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

<!-- Uncomment any of the following that are required -->
<!-- - [ ] Documentation -->
<!-- - [ ] ccloud/beginner-cloud -->
<!-- - [ ] ccloud/ccloud-stack -->
<!-- - [ ] ccloud-observability -->
<!-- - [ ] clickstream -->
<!-- - [ ] clients/avro -->
<!-- - [ ] clients/cloud -->
<!-- - [ ] cloud-etl -->
<!-- - [ ] connect-streams-pipeline -->
<!-- - [ ] cp-quickstart -->
<!-- - [ ] kubernetes/gke-base -->
<!-- - [ ] kubernetes/replicator-gke-cc -->
<!-- - [ ] microservices-orders -->
<!-- - [ ] multi-datacenter -->
<!-- - [ ] multiregion -->
<!-- - [ ] replicator-schema-translation -->
<!-- - [ ] replicator-security -->
<!-- - [ ] security/rbac -->


### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._

<!-- Uncomment any of the following that are required -->
<!-- - [ ] Documentation -->
<!-- - [ ] ccloud/beginner-cloud -->
<!-- - [ ] ccloud/ccloud-stack -->
<!-- - [ ] ccloud-observability -->
<!-- - [ ] clickstream -->
<!-- - [ ] clients/avro -->
<!-- - [ ] clients/cloud -->
<!-- - [ ] cloud-etl -->
<!-- - [ ] connect-streams-pipeline -->
<!-- - [ ] cp-quickstart -->
<!-- - [ ] kubernetes/gke-base -->
<!-- - [ ] kubernetes/replicator-gke-cc -->
<!-- - [ ] microservices-orders -->
<!-- - [ ] multi-datacenter -->
<!-- - [ ] multiregion -->
<!-- - [ ] replicator-schema-translation -->
<!-- - [ ] replicator-security -->
<!-- - [ ] security/rbac -->

ecukuroglu committed:
Update consumer_ccsr.py

The ordering of the variables is fixed to prevent deserialisation error.

ybyzek commented:
@confluentinc/clients , @rspurgeon : would you be able to review this PR?  Also, what changed such that this worked before but not now (i.e., why does "the deserialisation error" surface now but not before)?

rspurgeon commented:
@ecukuroglu Thank you for a submission to the example repository.  This PR makes a change that does not work for me, if you look at the definition of the `AvroDeserializer` type you can see that it expects the schema string and then the schema registry client:
```
def __init__(self, schema_str, schema_registry_client, from_dict=None):
```

Your change does not work for me with the expected exception:
```
Traceback (most recent call last):
  File "./consumer_ccsr.py", line 48, in <module>
    name_avro_deserializer = AvroDeserializer(schema_registry_client,
  File "/Users/rspurgeon/dev/confluentinc/examples/clients/cloud/python/ccloud-venv/lib/python3.8/site-packages/confluent_kafka/schema_registry/avro.py", line 273, in __init__
    self._reader_schema = parse_schema(loads(schema_str))
  File "/Users/rspurgeon/.pyenv/versions/3.8.6/lib/python3.8/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not SchemaRegistryClient
```

ecukuroglu commented:
@rspurgeon 
I am using python 3.8 too. It only works with the code below. I am using confluent_kafka-1.6.1 distribution. Which version do you use?
```
name_avro_deserializer = AvroDeserializer(schema_registry_client,
                                              ccloud_lib.name_schema,
                                              ccloud_lib.Name.dict_to_name)
count_avro_deserializer = AvroDeserializer(schema_registry_client,
                                               ccloud_lib.count_schema,
                                               ccloud_lib.Count.dict_to_count)
```
                                               
When I run the code below it throws me the error below:
```
       name_avro_deserializer = AvroDeserializer(ccloud_lib.name_schema,
                                              schema_registry_client,
                                              ccloud_lib.Name.dict_to_name)
    count_avro_deserializer = AvroDeserializer(ccloud_lib.count_schema,
                                               schema_registry_client,
                                               ccloud_lib.Count.dict_to_count)
```

```
   > python3 ./consumer_ccsr.py -f  $HOME/.confluent/librdkafka.config -t test2
Traceback (most recent call last):
  File "./consumer_ccsr.py", line 48, in <module>
    name_avro_deserializer = AvroDeserializer(ccloud_lib.name_schema,
  File "/Library/Python/3.8/site-packages/confluent_kafka/schema_registry/avro.py", line 278, in __init__
    self._reader_schema = parse_schema(loads(schema_str)) if schema_str else None
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not SchemaRegistryClient
```
This is the __init__ method for me:
```
    def __init__(self, schema_registry_client, schema_str,
                 to_dict=None, conf=None):
        self._registry = schema_registry_client
        self._schema_id = None
        # Avoid calling registry if schema is known to be registered
        self._known_subjects = set()
...
```

Based on the documentation schema_registry_client is prior to schema_str
https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#avrodeserializer

If you check the producer_ccsr.py (https://github.com/confluentinc/examples/blob/6.2.1-post/clients/cloud/python/producer_ccsr.py#L52) it also use schema_registry_client prior to schema_str which complies with https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#avroserializer

rspurgeon commented:
> I am using confluent_kafka-1.6.1 distribution. Which version do you use?

Thanks @ecukuroglu .  It appears the order of the parameters changed in v1.6.1. This example project's `requirements.txt` specifies 1.4.2 so it's not pulling the latest on my system.  We might want to fix this by using keyword arguments instead of the ordering so it works across the versions (at least for this issue). You'll notice that the Producer example you linked uses keyword arguments which avoids this issue in the consumer example.

ecukuroglu commented:
@rspurgeon I agree we should use keywords, it is safer. Thanks a lot

rspurgeon commented:
> @rspurgeon I agree we should use keywords, it is safer. Thanks a lot

@ecukuroglu If you'd like to post a change to this PR using keyword args, I can give it a test tomorrow. Otherwise I can write a fresh PR tomorrow.  Thanks for your help!

ecukuroglu committed:
Update consumer_ccsr.py

Updated the arguments using keywords from this link: https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#avrodeserializer

ecukuroglu commented:
@rspurgeon I updated the PR. Thanks

rspurgeon reviewed (approved):
LGTM, thank you for the contribution @ecukuroglu
